pykafka/producer.py

```diff
@@ -231,14 +231,32 @@ def _setup_owned_brokers(self):
                     self, partition.leader)
         return queued_messages
 
-    def stop(self):
-        """Mark the producer as stopped"""
-        self._running = False
+    def _stop_owned_brokers(self):
         self._wait_all()
         if self._owned_brokers is not None:
             for owned_broker in self._owned_brokers.values():
                 owned_broker.stop()
 
+    def stop(self, wait=True):
+        """Mark the producer as stopped, and wait until all messages to be sent
+
+        :param wait: whether we should wait until all messages to be sent or not
+        :type wait: bool
+        """
```

Comment:
This docstring would benefit from a detailed explanation of why the looped `join()` calls work here, similar to your comment on this pull request.